amads/time/meter/__init__.py

```diff
@@ -0,0 +1,619 @@
+
+
+    Parameters
+    ----------
+    pulse_lengths
+        Any valid list of pulse lengths, e.g., [4, 2, 1].
+    measure_length
+        Optional. If not provided it's taken to be given by the longest pulse length.
+    require_2_or_3_between_levels
+        Deafults to False. If True, raise a ValueError in the case of this condition not being met.
+
+    Returns
+    -------
+    list
+        Returns a list of lists with start positions by level.
+
+    Examples
+    --------
+
+    >>> qsl = starts_from_pulse_lengths(pulse_lengths=[4, 2, 1, 0.5])
+
+    >>> qsl[0]
+    [0.0, 4.0]
+
+    >>> qsl[1]
+    [0.0, 2.0, 4.0]
+
+    >>> qsl[2]
+    [0.0, 1.0, 2.0, 3.0, 4.0]
+
+    >>> qsl[3]
+    [0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0]
+
+    """
+
+    pulse_lengths = sorted(pulse_lengths)[::-1]  # largest number first
+
+    if not measure_length:
+        measure_length = float(pulse_lengths[0])
+
+    else:
+        if pulse_lengths[0] > measure_length:
+            raise ValueError(
+                f"The `pulse_length` {pulse_lengths[0]} is longer than the `measure_length` ({measure_length})."
+            )
+
+    if require_2_or_3_between_levels:
+        for level in range(len(pulse_lengths) - 1):
+            if pulse_lengths[level] / pulse_lengths[level + 1] not in [2, 3]:
+                raise ValueError(
+                    "The proportion between consecutive levels is not 2 or 3 in "
+                    f"this case: {pulse_lengths[level]}:{pulse_lengths[level + 1]}."
+                )
+
+    start_list = []
+
+    for pulse_length in pulse_lengths:
+        starts = starts_from_lengths(measure_length, pulse_length)
+        start_list.append(starts)
+
+    return start_list
+
+
+def coincident_pulse_list(
+    m: MetricalHierarchy,
+    granular_pulse: float,
+) -> list:
+    """
+    Create a flat list in the form [4, 1, 2, 1, 3, 1, 2, 1]
+    from the number of intersecting pulses at each position.
+
+    Parameters
+    --------
+    m
+        A level hierarchy, in the form of a list of list,
+        or a metrical hierarchy storing the same.
+    granular_pulse
+        The pulse value of the fastest level to consider e.g., 1, or 0.25.
+
+    Examples
+    --------
+
+    >>> m = [[0.0, 4.0], [0.0, 2.0, 4.0], [0.0, 1.0, 2.0, 3.0, 4.0]]
+    >>> coincident_pulse_list(m, granular_pulse=1)
+    [3, 1, 2, 1]
+
+    You can currently set the `granular_pulse` value to anything (this behavious may change).
+    For instance, a faster level that's not present simply pads the data out:
+
+    >>> m = [[0.0, 4.0], [0.0, 2.0, 4.0], [0.0, 1.0, 2.0, 3.0, 4.0]]
+    >>> coincident_pulse_list(m, granular_pulse=0.5)
+    [3, 0, 1, 0, 2, 0, 1, 0]
+
+    """
+    if isinstance(m, MetricalHierarchy):
+        m = m.start_hierarchy
+    elif not isinstance(m, list):
+        raise ValueError("The X must be a `MetricalHierarchy` object or a list")
+
+    measure_length = m[0][-1]
```

Comment:
Are we sure this is a good assumption to make? see above

Reply:
So this is consistent with current set up. If we refactor as above then this must change too. Heading to the issue. Only change for now is adding this as an attribute on the class.